### PR TITLE
Fix: The estimateGas function triggers an error when the density is e…

### PR DESCRIPTION
**Pull request by johanleroch**
…qual to 0 #234

### CODE REVIEW

File: packages/mangrove.js/src/market.ts [hunk 1]
```diff
@@ -591,18 +591,17 @@ class Market {
...
-    if (estimation.gt(MAX_MARKET_ORDER_GAS)) {
-      return BigNumber.from(MAX_MARKET_ORDER_GAS);
-    } else {
-      return estimation;
-    }
+    const maxMarketOrgerGas: BigNumber = BigNumber.from(MAX_MARKET_ORDER_GAS);
+    const estimation = density.isZero()
+      ? maxMarketOrgerGas
+      : offer_gasbase.add(volume.div(density));
+    return estimation.gt(MAX_MARKET_ORDER_GAS) ? maxMarketOrgerGas : estimation;
```
Comment (dontrolle) [inline packages/mangrove.js/src/market.ts:604]: Personally, I think something like the following for the last line is a bit more readable - but I'll leave this up to you.

```typescript
...
    if (estimation.lt(maxMarketOrgerGas))
      return estimation;

    return maxMarketOrgerGas;
```

**Review by dontrolle — CHANGES_REQUESTED** [review]
After settling the review-comments below, and the particular note that I think @adhusson or @jkrivine should just nod to the case for `density=0`, I think it's ready to go. :)

As a side-note - the `estimateGas()` func is going to be updated again when PR https://github.com/mangrovedao/mangrove/pull/231 lands. (I'll just rebase, when it's reviewed.)

File: packages/mangrove.js/src/market.ts [hunk 2]
```diff
@@ -591,18 +591,17 @@ class Market {
...
         : await this.#bidsSemibook.getRawConfig();
-    const estimation = rawConfig.local.offer_gasbase.add(
-      volume.div(rawConfig.local.density)
-    );
-    if (estimation.gt(MAX_MARKET_ORDER_GAS)) {
-      return BigNumber.from(MAX_MARKET_ORDER_GAS);
-    } else {
-      return estimation;
-    }
+    const maxMarketOrgerGas: BigNumber = BigNumber.from(MAX_MARKET_ORDER_GAS);
```
[the remark this answers is not in the document]
Comment (johanleroch) [reply packages/mangrove.js/src/market.ts:600]: oups! thanks :)

File: packages/mangrove.js/src/market.ts (hunk 1 above)
Comment (johanleroch) [reply packages/mangrove.js/src/market.ts:604]: I agree, more readable like this :)

**Review by dontrolle — APPROVED** [review]

**Pull request merged**